apis/apps/v1beta4/emqxrebalance_types.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta4
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EmqxRebalanceSpec defines the desired spec of EmqxRebalance
+type EmqxRebalanceSpec struct {
+	// +kubebuilder:validation:required
+	InstanceName string `json:"instanceName,omitempty"`
+	// +kubebuilder:validation:required
+	RebalanceStrategy *RebalanceStrategy `json:"rebalanceStrategy,omitempty"`
+}
+
+type RebalanceStrategy struct {
+	// ConnEvictRate represents the source node client disconnect rate per second.
+	ConnEvictRate int32 `json:"connEvictRate,omitempty"`
+	// SessEvictRate represents the source node session evacuation rate per second.
+	SessEvictRate int32 `json:"sessEvictRate,omitempty"`
+	// WaitTakeover represents the time in seconds to wait for a client to
+	// reconnect to take over the session after all connections are disconnected.
+	WaitTakeover int32 `json:"waitTakeover,omitempty"`
+	// WaitHealthCheck represents the time (in seconds) to wait for the LB to
+	// remove the source node from the list of active backend nodes. After the
+	// specified waiting time is exceeded,the rebalancing task will start.
+	WaitHealthCheck int32 `json:"waitHealthCheck,omitempty"`
+	// AbsConnThreshold represents the absolute threshold for checking
+	// connection balance.
+	AbsConnThreshold int32 `json:"absConnThreshold,omitempty"`
+	// RelConnThreshold represents the relative threshold for checking
+	// connection balance.
+	RelConnThreshold string `json:"relConnThreshold,omitempty"`
```

Comment:
string or float64 ?

Reply:
> string or float64 ?

string is better, kubebuilder not recommend float

Reply:
But if use string, user can be write any string, it will be crash ?